Respect style containment on list items

rwlbuis opened the pull request:
#### cbe37d744f2fd617ae9dfa317f3e553c02b32c93
<pre>
Respect style containment on list items
<a href="https://bugs.webkit.org/show_bug.cgi?id=256884">https://bugs.webkit.org/show_bug.cgi?id=256884</a>

Reviewed by Oriol Brufau.

List items can be affected by style containment
as a scope for the current counter, just like
normal lists [1].
[1] <a href="https://drafts.csswg.org/css-contain-2/#containment-style">https://drafts.csswg.org/css-contain-2/#containment-style</a>

* LayoutTests/TestExpectations:
* Source/WebCore/rendering/RenderListItem.cpp:
(WebCore::enclosingList):

Canonical link: <a href="https://commits.webkit.org/264273@main">https://commits.webkit.org/264273@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/83063a9f034fd1764f0cc92af3e8701ff2ad254a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7217 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7465 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7644 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8836 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7434 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/7227 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/8801 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7398 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/10332 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7345 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/8039 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/6648 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8943 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/5390 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/6567 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/14301 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/7029 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/6670 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/9543 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7152 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/5848 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6511 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1707 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10711 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6894 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Loirooriol reviewed (approved):
Looking at https://chromium-review.googlesource.com/c/chromium/src/+/4161712/2, Blink added the check in `IsList`, which is used for both `EnclosingList` and `NextListItem`.

You are only doing the former, I don't remember if the latter was actually needed in some case. But well at least this is an improvement, so I guess it's fine, it's not like Blink handles all the details correctly anyways.

rwlbuis committed:
Respect style containment on list items
https://bugs.webkit.org/show_bug.cgi?id=256884

Reviewed by Oriol Brufau.

List items can be affected by style containment
as a scope for the current counter, just like
normal lists [1].
[1] https://drafts.csswg.org/css-contain-2/#containment-style

* LayoutTests/TestExpectations:
* Source/WebCore/rendering/RenderListItem.cpp:
(WebCore::enclosingList):

Canonical link: https://commits.webkit.org/264273@main
